### PR TITLE
Show full row content of problematic entries in import-issue list (#91 part 1)

### PR DESCRIPTION
**Pull request by StepKie**
## Summary

Part 1 of #91. The "Correct this entry" Scryfall picker (part 2) is deferred to a follow-up PR — see #91 for scope.

**Before:** issue list showed only the *reason* + a comma list of row numbers. To see what was actually in row 3827 the user had to hop into a CSV editor.

**After:** each issue group lists every affected row as `row N: <raw csv content>` so the offending line is right there in the panel.

| | |
| --- | --- |
| `MtgCsvHelper/Enrichment/ParsedRow.cs` | Added `string? RawContent` to the record. Populated in `MtgCardCsvHandler` alongside the in-memory card. |
| `SetInfoEnricher.cs`, `CatalogValidator.cs`, `CardmarketIdEnricher.cs` | Pass `row.RawContent` to every `new ImportIssue(...)` so the per-row display has something to render for the post-parse failures. (Previously only parse-time issues — `Count must be positive`, `Invalid value for column X` — had it.) |
| `MtgCsvProcessor.razor` | Per-issue group now lists each affected row as `row N: <raw>` with the raw line inside a `<code>` block. Caps at 25 rows per group with "... and N more" caption. Guards against an empty `<code>` when `RawContent` is null. |
| `MtgCsvProcessor.razor.css` (new) | Scoped via Blazor CSS isolation. `.issue-row-list` is muted with `--mud-palette-text-secondary`; `.issue-row-raw` inherits color from the parent list item so warning/error rows keep their existing tint. Monospace for the raw content. |
| `MaxIssueRowsShown` | Hoisted to a `const` at the top of `@code` instead of inline in the `@foreach`. |

The early iteration of this PR had a regex-based highlight layer that wrapped offending values in `<mark>` tags + a separate `IssueHighlights` static class + 6 unit tests for it. After previewing the styled result locally, that whole layer was dropped — the raw row alone was good enough at this scope.

## Test plan

- [x] `dotnet test MtgCsvHelper.slnx` — 173 passing
- [x] `dotnet build` — clean, 0 warnings
- [x] Local dev-server verification: post-parse issues (MB1 / EMN / token name mismatches from the sample collections) now show the raw CSV row beneath the row number ✅
- [ ] CI green on this push

## Related issues (filed during diagnosis, with explicit permission)

- #92 — Catalog uses Scryfall `default_cards` so reprint set codes (MB1 etc.) don't resolve
- #93 — Dragon Shield decorates token names with parentheticals we don't strip
- #94 — Round-trip DFC names lose their back face when the front face is shared by multiple printings

All three are out of scope for this PR.

### CODE REVIEW

File: MtgCsvHelper.BlazorWebAssembly/Pages/MtgCsvProcessor.razor [hunk 1]
```diff
@@ -237,16 +237,27 @@
...
-									<MudChip T="string" Size="Size.Small" Variant="Variant.Outlined">@g.Count() row@(g.Count() == 1 ? "" : "s")</MudChip>
+									<MudChip T="string" Size="Size.Small" Variant="Variant.Outlined">@issues.Count row@(issues.Count == 1 ? "" : "s")</MudChip>
 								</MudStack>
-								<MudText Typo="Typo.caption" Class="mud-text-secondary">@(rows.Count == 1 ? "row" : "rows"): @preview</MudText>
+								<div class="issue-row-list">
+									@foreach (var issue in issues.Take(MaxIssueRowsShown))
+									{
+										<div class="issue-row">
+											row @issue.RowNumber@(string.IsNullOrEmpty(issue.RawContent) ? "" : ": ")<code class="issue-row-raw">@issue.RawContent</code>
+										</div>
```
[the remark this answers is not in the document]
Comment (StepKie) [reply MtgCsvHelper.BlazorWebAssembly/Pages/MtgCsvProcessor.razor:255]: ✅ Done in 08d3f4f: PR description rewritten to match the final shape (highlight layer is gone, raw row is plain). And the empty `<code></code>` is now guarded — switched the inline ternary to a `@if` block so the suffix only renders when RawContent is present.

File: MtgCsvHelper.BlazorWebAssembly/Pages/MtgCsvProcessor.razor.css [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+	color: var(--mud-palette-text-secondary);
+	font-size: 0.85rem;
+}
+
+.issue-row {
+	word-break: break-word;
+}
+
+.issue-row-raw {
+	font-family: ui-monospace, "Cascadia Code", Menlo, Consolas, monospace;
```
[the remark this answers is not in the document]
Comment (StepKie) [reply MtgCsvHelper.BlazorWebAssembly/Pages/MtgCsvProcessor.razor.css:20]: ❌ Skipped: highlighting was deliberately dropped from this PR after a local preview — `<mark>` is no longer emitted anywhere, so an `.issue-row-raw mark { ... }` rule would be dead CSS. If a follow-up brings highlighting back, the rule should land in the same commit so the styling and the markup arrive together.

File: MtgCsvHelper/MtgCardCsvHandler.cs [hunk 3]
```diff
@@ -85,10 +86,10 @@ public async Task<ParseResult> ParseCollectionCsvAsync(Stream csvStream, Cancell
 						IssueSeverity.Error,
 						rowNum,
 						$"Count must be positive (got {card.Count})",
-						RawContent: csv.Parser.RawRecord?.TrimEnd()));
+						RawContent: rawContent));
 					continue;
 				}
-				rows.Add(new ParsedRow(card, rowNum));
+				rows.Add(new ParsedRow(card, rowNum, rawContent));
```
[the remark this answers is not in the document]
Comment (StepKie) [reply MtgCsvHelper/MtgCardCsvHandler.cs:92]: ✅ Done in 881b800. Added RawContent assertions to each existing FaultToleranceTests path that exercises the enricher pipeline (UnknownSetCode covers both the SetInfoEnricher warning and the CatalogValidator error via AllSatisfy; NameDoesNotMatch, FoilOnNonFoil, SetAndCollectorNotInCatalog each get a one-liner). Plus a new test `NonPositiveCount_RaisesError_RowDropped_WithRawContent` for the parse-time capture path in MtgCardCsvHandler (Count <= 0 emits its issue inside the GetRecord try-block, outside the enricher pipeline — would have been missed if I'd only asserted on the enricher paths). 174 passing; Cardmarket's enricher path stays uncovered because targeting it would mean mocking the resolver + network for what's currently a low regression risk.

File: MtgCsvHelper.BlazorWebAssembly/Pages/MtgCsvProcessor.razor [hunk 4]
```diff
@@ -237,16 +239,33 @@
 						@foreach (var g in grouped)
 						{
 							var (severity, reason) = g.Key;
-							var rows = g.Select(i => i.RowNumber).OrderBy(rowNum => rowNum).ToList();
-							var preview = string.Join(", ", rows.Take(8));
-							if (rows.Count > 8) preview += $", … (+{rows.Count - 8} more)";
+							var issues = g.OrderBy(i => i.RowNumber).ToList();
+							var highlights = ExtractHighlightValues(reason);
+							const int MaxRowsShown = 25;
```
[the remark this answers is not in the document]
Comment (StepKie) [reply MtgCsvHelper.BlazorWebAssembly/Pages/MtgCsvProcessor.razor]: ✅ Done in 54bb2cf: hoisted to a `MaxIssueRowsShown` const at the top of `@code` alongside the other static fields. (Subsequently kept through the rebase — the const survived the highlight-layer removal in f3e1bfc.)

File: MtgCsvHelper.BlazorWebAssembly/Pages/MtgCsvProcessor.razor [hunk 5]
```diff
@@ -349,6 +368,39 @@
...
+	// contains an HTML-meaningful char and would mismatch the escaped text. Returns a
+	// MarkupString so Blazor renders the <mark> tags rather than printing them literally.
+	static MarkupString HighlightValues(string raw, IReadOnlyList<string> values)
+	{
+		var encoded = WebUtility.HtmlEncode(raw);
+		foreach (var v in values)
+		{
+			if (string.IsNullOrEmpty(v)) { continue; }
+			var ev = WebUtility.HtmlEncode(v);
+			encoded = encoded.Replace(ev, $"<mark>{ev}</mark>", StringComparison.Ordinal);
```
[the remark this answers is not in the document]
Comment (StepKie) [reply MtgCsvHelper.BlazorWebAssembly/Pages/MtgCsvProcessor.razor]: ✅ Initially addressed in 54bb2cf with `OrderByDescending(s => s.Length)` + a regression test pinning the longest-first contract. Then **moot**: in f3e1bfc the whole highlight layer was dropped after a local preview (user preferred raw row without `<mark>` wraps), so this code path no longer exists. Good catch though — if highlighting comes back, the length-sorted iteration is the right contract.

File: MtgCsvHelper.BlazorWebAssembly/Pages/MtgCsvProcessor.razor [hunk 6]
```diff
@@ -349,6 +368,39 @@
...
+		foreach (Match m in _quotedValue.Matches(reason)) { values.Add(m.Groups[1].Value); }
+		foreach (Match m in _hashedValue.Matches(reason)) { values.Add(m.Groups[1].Value); }
+		return [.. values.Distinct()];
+	}
+
+	// HTML-escapes the raw row, then wraps every occurrence of every highlight value with
+	// <mark>. Replacing post-escape avoids the case where the highlight value itself
+	// contains an HTML-meaningful char and would mismatch the escaped text. Returns a
+	// MarkupString so Blazor renders the <mark> tags rather than printing them literally.
+	static MarkupString HighlightValues(string raw, IReadOnlyList<string> values)
```
[the remark this answers is not in the document]
Comment (StepKie) [reply MtgCsvHelper.BlazorWebAssembly/Pages/MtgCsvProcessor.razor]: ✅ Initially addressed in 54bb2cf: extracted helpers into `MtgCsvHelper.IssueHighlights` (separate static class so the tests don't need Blazor scaffolding) + 6 xUnit tests covering happy path, no-captures, empty values, HTML-significant chars, longest-first ordering. Then **moot**: the helper class + its tests were removed in f3e1bfc when the highlight layer was dropped.

File: MtgCsvHelper/Resources/SampleCsvs/Collection/moxfield-haves-collection.csv [hunk 7]
```diff
@@ -0,0 +1,7386 @@
+"Count","Tradelist Count","Name","Edition","Condition","Language","Foil","Tags","Last Modified","Collector Number","Alter","Proxy","Purchase Price"
```
[the remark this answers is not in the document]
Comment (StepKie) [reply MtgCsvHelper/Resources/SampleCsvs/Collection/moxfield-haves-collection.csv]: ✅ Deferred — the file is gone from this PR (removed in 54bb2cf). Author decision per the personal-data concern and the "unused fixture" point. Will resurface in #94's round-trip test once that lands, likely with prices + timestamps stripped.

File: MtgCsvHelper/IssueHighlights.cs [hunk 8]
```diff
@@ -0,0 +1,48 @@
...
+/// be unit-tested directly. The page wraps <see cref="HighlightAsHtml"/>'s string output in
+/// a <c>MarkupString</c> before rendering.
+/// </summary>
+public static partial class IssueHighlights
+{
+	// Two patterns cover every emit site in the parser pipeline:
+	//   - 'quoted'  — set codes, card names, column names ("Set code 'MB1' not found", "Invalid value 'X' for column 'Y'")
+	//   - #hashed   — collector numbers ("No printing at MB1 #805 in Scryfall data")
+	[GeneratedRegex(@"'([^']+)'", RegexOptions.Compiled)]
+	private static partial Regex QuotedValue();
```
[the remark this answers is not in the document]
Comment (StepKie) [reply MtgCsvHelper/IssueHighlights.cs]: ✅ Moot — `IssueHighlights` was removed entirely in f3e1bfc when the highlight layer was dropped from this PR. Noted for any future regex work in this repo: `RegexOptions.Compiled` is redundant with `[GeneratedRegex]` (source-gen already emits a compiled implementation).

File: MtgCsvHelper/IssueHighlights.cs [hunk 9]
```diff
@@ -0,0 +1,48 @@
...
+	[GeneratedRegex(@"#(\S+)", RegexOptions.Compiled)]
+	private static partial Regex HashedValue();
```
[the remark this answers is not in the document]
Comment (StepKie) [reply MtgCsvHelper/IssueHighlights.cs]: ✅ Moot — same as the sibling finding on the other `[GeneratedRegex]` declaration. `IssueHighlights` was deleted in f3e1bfc.

File: MtgCsvHelper/IssueHighlights.cs [hunk 10]
```diff
@@ -0,0 +1,48 @@
...
+	/// wrap when handing it to Blazor.
+	/// </summary>
+	public static string HighlightAsHtml(string raw, IReadOnlyList<string> values)
+	{
+		var encoded = WebUtility.HtmlEncode(raw);
+		foreach (var v in values.OrderByDescending(s => s.Length))
+		{
+			if (string.IsNullOrEmpty(v)) { continue; }
+			var ev = WebUtility.HtmlEncode(v);
+			encoded = encoded.Replace(ev, $"<mark>{ev}</mark>", StringComparison.Ordinal);
```
[the remark this answers is not in the document]
Comment (StepKie) [reply MtgCsvHelper/IssueHighlights.cs]: ✅ Moot — `HighlightAsHtml` was deleted along with the rest of `IssueHighlights` in f3e1bfc. The tag-corruption observation is real for any future revival of this code: a regex-tokeniser pass (split into highlight + non-highlight segments, then concat) would be the robust shape.
